modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -194,13 +200,24 @@ func (h *HTTPTransport) NewTransport(ctx caddy.Context) (*http.Transport, error)
 				network = dialInfo.Network
 				address = dialInfo.Address
 			}
+
 			conn, err := dialer.DialContext(ctx, network, address)
 			if err != nil {
 				// identify this error as one that occurred during
 				// dialing, which can be important when trying to
 				// decide whether to retry a request
 				return nil, DialError{err}
 			}
+
+			// if read/write timeouts are configured and this is a TCP connection, enforce the timeouts
+			if tcpConn, ok := conn.(*net.TCPConn); ok && (h.ReadTimeout > 0 || h.WriteTimeout > 0) {
```

Comment:
Hmm, won't this mean that if a read timeout is configured but not a write timeout, then it will _always_ timeout _immediately_ on write, for example? Cause it'll call `c.TCPConn.SetDeadline(time.Now())` effectively.

Reply:
Ah yep, I just implemented this real quick on my way out the door. I will fix that. Thanks!